routes/taskHandler.js

```diff
@@ -10,16 +10,24 @@ taskHandler.getTasks = function(req, res, next) {
         .catch((err) => next(err));
 };
 
-taskHandler.createTask = async function (req, res, next) {
-  const { title, isCompleted, notes } = req.body;
+taskHandler.createOrUpdateTask = async function (req, res, next) {
 
   Task.findOneAndUpdate(
-    { title: title },
-    { $set: { isCompleted: isCompleted, notes: notes, owner: req.user.email } },
+    req.body.id ? { _id: req.body.id } 
+                : { _id: new mongoose.mongo.ObjectId() },
```

Comment:
Did it not auto assign an ID on create?

Also, you could to this a bit more succinctly as `{ _id: req.body?.id ?? new mongoose.Types.ObjectId() }`

That way if it's `null` or `undefined`, it'll use the mongoose ID. Also, you would likely see an error in the event `id` isn't passed without the optional chaining as`?.id`.

However, ideally we wouldn't look for id from body, we would look at `req.params.id` with a /tasks/:id on the  patch endpoint. Otherwise, we have two identical endpoints, just different methods. 

Also, I think we want to have some permission checking in here as well.

So this line could be changed to: `{ _id: req.params?.id ?? new mongoose.Types.ObjectId(), owner: req.user.email }`. That would ensure that this is only changed on a patch, and only with the authorized user.